# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1260425561 18000
#      Thu Dec 10 01:12:41 2009 -0500
# Node ID 225bfa546ae7a7403fafba3406f8745f23c68fb2
# Parent  7cd2e1b372e577e23b44ced03336969ce1815cb0
allow + at end of fopen modestrings like rb+

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2009-12-10  John W. Eaton  <jwe@octave.org>
+
+	* file-io.cc (fopen_mode_to_ios_mode): Allow "+" at the end of
+	modestrings like "rb+".
+
 2009-12-10  John W. Eaton  <jwe@octave.org>
 
 	* oct-stream.cc (BEGIN_C_CONVERSION, BEGIN_S_CONVERSION):
 	Store characters directly in appropriately sized std::string object.
 	(FINISH_CHARACTER_CONVERSION): Do store ASCII NUL values.
 
 2009-12-09  John W. Eaton  <jwe@octave.org>
 
diff --git a/src/file-io.cc b/src/file-io.cc
--- a/src/file-io.cc
+++ b/src/file-io.cc
@@ -179,34 +179,34 @@ fopen_mode_to_ios_mode (const std::strin
       if (! error_state)
 	{
 	  if (mode == "rt")
 	    retval = std::ios::in;
 	  else if (mode == "wt")
 	    retval = std::ios::out | std::ios::trunc;
 	  else if (mode == "at")
 	    retval = std::ios::out | std::ios::app;
-	  else if (mode == "r+t")
+	  else if (mode == "r+t" || mode == "rt+")
 	    retval = std::ios::in | std::ios::out;
-	  else if (mode == "w+t")
+	  else if (mode == "w+t" || mode == "wt+")
 	    retval = std::ios::in | std::ios::out | std::ios::trunc;
-	  else if (mode == "a+t")
+	  else if (mode == "a+t" || mode == "at+")
 	    retval = std::ios::in | std::ios::out | std::ios::app;
 	  else if (mode == "rb" || mode == "r")
 	    retval = std::ios::in | std::ios::binary;
 	  else if (mode == "wb" || mode == "w")
 	    retval = std::ios::out | std::ios::trunc | std::ios::binary;
 	  else if (mode == "ab" || mode == "a")
 	    retval = std::ios::out | std::ios::app | std::ios::binary;
-	  else if (mode == "r+b" || mode == "r+")
+	  else if (mode == "r+b" || mode == "rb+" || mode == "r+")
 	    retval = std::ios::in | std::ios::out | std::ios::binary;
-	  else if (mode == "w+b" || mode == "w+")
+	  else if (mode == "w+b" || mode == "wb+" || mode == "w+")
 	    retval = (std::ios::in | std::ios::out | std::ios::trunc
 		      | std::ios::binary);
-	  else if (mode == "a+b" || mode == "a+")
+	  else if (mode == "a+b" || mode == "ab+" || mode == "a+")
 	    retval = (std::ios::in | std::ios::out | std::ios::app
 		      | std::ios::binary);
 	  else
 	    ::error ("invalid mode specified");
 	}
     }
 
   return retval;
